USSF-ORBIT/ussf-portal-client
# refactor: Implement dnd-kit

# SC-1952
This PR has been hijacked by :star: @jcbcapps :star:


## Proposed changes
Replace `react-beautiful-dnd` with `dnd-kit`

Todo:
- ~Update e2e test~
- ~Fix add and remove bookmark page refresh~


---



## Reviewer notes
How to test:
- Run the app locally
- In a user's My Space, confirm that you can drag and drop bookmarks
- In another tab, navigate to `localhost:8888` and confirm that the new order is represented in the database
- Test the ability to add and remove bookmarks in My Space
- Check out the e2e test here: https://github.com/USSF-ORBIT/ussf-portal/pull/277


## Setup



### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-cms
yarn dev
```

Login to the portal http://localhost:3000

### Start storybook

```sh
yarn storybook
```

Login to storybook http://localhost:6006, though the command above should open it for you

---

## Code review steps

### As the original developer, I have

- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

## Screenshots
